CodegenProposal.md

```diff
@@ -338,13 +338,142 @@ A `SchemaConfiguration` object will also be generated for your schema. This obje
+case .__unknown(_): break
+default: break
+}
+```
+Because pattern matching is being used to match against the underling `EnumType` cases, you must also provide a default case.
+
+To ensure exhaustive switch cases without a default case your generated cases can be wrapped in `.case()`.
+```swift
+switch size {
+case .case(.SMALL): break
+case .case(.AVERAGE): break
+case .case(.LARGE): break
+case .__unknown(_): break
+}
+```
+If you want to ignore the unknown case, you can access the `.value` field, which returns an optional value of the wrapped type. If the type is an unknown case `.value` will be `nil`.
+```swift
+switch size.value {
+case .SMALL: break
+case .AVERAGE: break
+case .LARGE: break
+default: break 
+// or
+case .none: break
+}
+```
+See [GraphQLEnum.swift](Sources/ApolloAPI/GraphQLEnum.swift) for implementation details.
+
 # `InputObject` Generation
 
-TODO
+Input objects will be generated for each `input` type in the schema that is used in an argument for any of the operations defined in your application. Input objects are structs that are backed by a `InputDict` struct that stores the values for the fields on the input object in a dictionary. This allows for `InputObject`s to be treated as values types but use copy-on-write semantics under the hood.
 
-# `EnumType` Generation
+Nullable fields on input objects are represented using `GraphQLNullable` to allow for both `null` and `nil` values. 
 
-TODO
+Following the [Input Coercion rules](https://spec.graphql.org/draft/#sec-Input-Objects.Input-Coercion) from the GraphQL spec, the server defined default value for a field will be used when passing `nil`. For non-nullable fields, if the schema provides a default value, the field will be represented as an optional to allow for this. Nullable fields on input objects are represented using `GraphQLNullable` to allow for both `null` and `nil` values.
+
+**Examples:**
+
+Nullable field with no default value:
+```
+input MyInput {
+  size: RelativeSize
+}
+```
+```swift
+struct MyInput: InputObject {
+  public private(set) var dict: InputDict
+
+  init(size: GraphQLNullable<RelativeSize> = nil) {
+    dict = InputDict(["size": size])
+  }
+
+  var size: GraphQLNullable<RelativeSize> {
+    get { dict["size"] }
+    set { dict["size"] = newValue }
+  }
+}
+```
+Nullable field with a default value:
+```
+input MyInput {
+  size: RelativeSize = SMALL
+}
+```
+```swift
+struct MyInput: InputObject {
+  public private(set) var dict: InputDict
+
+  init(size: GraphQLNullable<RelativeSize>) { ... }
+
+  /// If `.none`, defaults to server-provided default value (.SMALL)
+  var size: GraphQLNullable<RelativeSize> { ... }
+}
+```
+Non-nullable field with no default value:
+```
+input MyInput {
+  size: RelativeSize!
+}
+```
+```swift
+struct MyInput: InputObject {
+  public private(set) var dict: InputDict
+
+  init(size: RelativeSize) { ... }
+  
+  var size: RelativeSize { ... }
+}
+```
+Non-nullable field with a default value:
+```
```

Comment:
probably worth tagging some of these operation and schema examples with `graphql` so it renders prettier

Reply:
I don't think that graphql works in GitHub markdown, does it? I will add that!

Reply:
I believe that it does

Reply:
You are correct! This is done and merged into the release 1.0 branch now!